### PR TITLE
feat: fix key-value-image + add test

### PR DESCRIPTION
**Pull request by JFKFred**

### CODE REVIEW

File: core/validator/src/rules/zod.ts [hunk 1]
```diff
@@ -8,7 +8,8 @@ import z, { type RefinementCtx } from "zod";
 const HEX56_REGEXP = /^[0-9a-fA-F]{56}$/;
 const REGEX_MEDIA_TYPE = /^(image\/|video\/|audio\/|application\/|model\/)/;
 const URI_REGEXP = /^(https?|ftp|ipfs):\/\/[^\s/$.?#].[^\s]*$/i;
-const BASE64_REGEXP = /^[A-Za-z0-9+/=]+$/;
+const BASE64_REGEXP =
```
Comment (mrabdibdi-anvil) [inline core/validator/src/rules/zod.ts]: I think I would keep the BASE64 as a specific regex.
and make another one with your data prefix

Comment (JFKFred) [reply core/validator/src/rules/zod.ts]: When to use witch one ?
I might don't understand when to use it

Comment (mrabdibdi-anvil) [reply core/validator/src/rules/zod.ts]: If you do not use it anywhere you can ignore my comment

Comment (JFKFred) [reply core/validator/src/rules/zod.ts]: Nope I don't and the new Regex is the format needed on cardano.

https://docs.nmkr.io/nmkr-studio/token/metadata/fully-on-chain-nfts 👀

**Review by mrabdibdi-anvil — APPROVED** [review]

**Pull request merged**